draft/XXXX-multinetjs-autopublish/README.md

```diff
@@ -0,0 +1,86 @@
+---
+RFC: XXXX
+Author: Roni Choudhury
+Status: draft
+Created: 2020-07-09
+Last-Modified: 2020-07-09
+---
+
+# RFC XXXX: Autopublishing for MultinetJS
+
+One of the active pieces of software driving the Multinet project is
+`multinetjs`, a TypeScript library providing clientside bindings and utility
+functions for the Multinet API. This software grows with the needs of the
+project, specifically of the [Multinet
+client](https://github.com/multinet-app/multinet-client). This means that often,
+a new feature in the client requires a new feature in MultinetJS, and to support
+this use case, this RFC lays out a plan for more easily publishing prerelease
+and development versions of the library for use in client development.
+
+## Motivation
+
+MultinetJS does not currently have an automated, official release process.
+Releases are done by hand after a pull request is approved, and before that step
+happens, developers are expected to use `yarn link` to provide development
+linkages between MultinetJS and the client codebase.
+
+While this procedure works well in local development, it leaves CI out. In
+particular, since the Netlify build previews of the client rely on published
+versions of MultinetJS, build previews depending on a development version of
+MultinetJS simply cannot be built.
+
+A procedure for automated or semi-automated publishing of MultinetJS, even for
+development versions, would fix this issue. It would also eliminate the manual
+step currently needed to perform releases to NPM.
+
+## Proposals
+
+The proposals here use GitHub Actions to drive publishing pipelines to NPM on
+both pull request branches, and on master. Details for each follow.
+
+### Development Releases
+
+The idea here is to automatically publish development versions based on a PR
+branch to NPM automatically using CI. On each push from a PR branch, the GH
+Actions workflow can run the tests, perform a build, and if successful, publish
+a prerelease version to NPM. The version number can be taken from the current
+value in `package.json`, appended with the first six characters of the current
+commit's Git SHA--for example `multinet-0.47.3-a1c0ef` ([Clause
+9](https://semver.org/#spec-item-9) of the semver spec stipulates that
+prerelease versions can be indicated by appending alphanumeric strings in this
+manner).
+
+Alternatively, instead of automatically pushing these releases, they could
+instead be done manually via an NPM script. However, this would have the
+drawback of requiring distribution of credentials to the different developers.
```

Comment:
You mention credentials here, but not above. In the above case, with the CI doing the pushing, are the credentials stored in the GH secrets, and, thus, don't need to be distributed?

Reply:
Yes, and I will update the text to mention GH secrets.

Reply:
See e5e6ff82366f617882361678e4889656c32dea1c.